test/LinearSolvers/smallsystem.jl

```diff
@@ -0,0 +1,44 @@
+using Test
+using CLIMA
+using CLIMA.LinearSolvers
+using CLIMA.GeneralizedConjugateResidualSolver
+
+using LinearAlgebra, Random
+
+
+# this test setup is partly based on IterativeSolvers.jl, see e.g
+# https://github.com/JuliaMath/IterativeSolvers.jl/blob/master/test/cg.jl
+@testset "LinearSolvers small full system" begin
+  n = 10
+
+  expected_iters = Dict(Float32 => 4, Float64 => 6)
+
+  for T in [Float32, Float64]
+    Random.seed!(44)
+
+    A = rand(T, n, n)
```

Comment:
Should we avoid `rand` in the tests?

Reply:
I think it is okay to use `rand` as long as we are setting the seed.